modules/apps/segments/segments-api/src/main/java/com/liferay/segments/manager/SegmentsExperienceManager.java

```diff
@@ -47,8 +50,25 @@ public long getSegmentsExperienceId(HttpServletRequest httpServletRequest) {
 			(ThemeDisplay)httpServletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
+		long plid = 0;
+
+		if (themeDisplay != null) {
+			plid = themeDisplay.getPlid();
+		}
+		else {
+			Map<String, String[]> parameterMap =
```

Comment:
What is the use case we don't have a themeDisplay but we have a plid in the URL?

Reply:
If there is a use case, why don't you just need ParamUtil.getLong(httpServletRequest, "plid")?

Reply:
For the first question, in some cases, for example, for the LayoutStructureCommonStylesCSSServlet is called before ThemeDisplay is setted in the request and we have the plid as a parameter.

For the second question, yes you are right, I can use ParamUtil.

Reply:
And the LayoutStructureCommonStylesCSSServlet call the SegmentExperienceManager?

Reply:
The doubt I have is that why we don't have a problem now, because we are calling getSegmentsExperienceId, in the  LayoutStructureCommonStylesCSSServlet without reading the parameter plid.

Do we have associated with this behaviour?

Reply:
@cgoncas, we don't have the problem now because we always return the parameter from the URL 😭

Reply:
@jkappler and @Mikellorza could you debug the code with me for a second, so I understand correctly why this is not failing now?

Reply:
After discussion with @jkappler and @Mikellorza we have decided that instead take the plid from the URL, maybe we should address the use case where the ServicePreAction is not being executed, because we are returning the defaultSegmentExperienceId and this is not the expected behaviour.